src/Chainweb/Payload/RestAPI.hs

```diff
@@ -142,6 +148,25 @@ payloadPostApi
     . Proxy (PayloadPostApi v c)
 payloadPostApi = Proxy
 
+-- | @POST \/chainweb\/\<ApiVersion\>\/\<InstanceId\>\/chain\/\<ChainId\>\/heightedpayload\/batch@
+--
+-- The query may return any number (including none) of the requested payload
+-- data. Results are returned in any order.
+--
+type HeightedPayloadPostApi_
+    = "payload"
+    :> "batch"
+    :> ReqBody '[JSON] [(BlockHeight, BlockPayloadHash)]
+    :> Post '[JSON] [PayloadData]
```

Comment:
I wonder if we should include the block height also in the response. E.g. analogous to the request as`[(Blockheight, PayloadData)]` (or whatever we end up using, cf. comment above).